src/status_im/wallet/choose_recipient/core.cljs

```diff
@@ -68,18 +65,67 @@
             ethereum/default-transaction-gas))
 
 (re-frame/reg-fx
- :resolve-address
+ ::resolve-address
  (fn [{:keys [registry ens-name cb]}]
    (ens/get-addr registry ens-name cb)))
 
+(re-frame/reg-fx
+ ::resolve-addresses
+ (fn [{:keys [registry ens-names callback]}]
+   ;; resolve all addresses then call the callback function with the array of 
+   ;;addresses as parameter
+   (-> (js/Promise.all
+        (clj->js (mapv (fn [ens-name]
+                         (js/Promise.
+                          (fn [resolve reject]
+                            (ens/get-addr registry ens-name resolve))))
+                       ens-names)))
+       (.then callback)
+       (.catch (fn [error]
+                 (js/console.log error))))))
+
+(fx/defn resolve-ens-addresses
+  {:events [:wallet.send/resolve-ens-addresses]}
+  [{{:networks/keys [current-network] :wallet/keys [all-tokens] :as db} :db}
+   message origin address]
+  ;; first we get a vector of ens-names to resolve and a vector of paths of
+  ;; these names
+  (let [{:keys [paths ens-names]}
+        (reduce (fn [acc path]
+                  (let [address (get-in message path)]
+                    (if (ens/is-valid-eth-name? address)
+                      (-> acc
+                          (update :paths conj path)
+                          (update :ens-names conj address))
+                      acc)))
+                {:paths [] :ens-names []}
+                [[:address] [:function-arguments :address]])]
+    (if (empty? ens-names)
+      ;; if there is no ens-names, we dispatch request-uri-parsed immediately
+      (re-frame/dispatch [:wallet/request-uri-parsed message origin])
```

Comment:
we shouldn't use `dispatch` here, call of `(request-uri-parsed cofx ...` should be here

Reply:
@yenda Again, please confirm but this section occurs after the `parse-uri-handler` call has occurred so we're just resolving any ENS names and then updating the `send-transaction` details.

Reply:
again, let's wait for @yenda :)  but if you want to dispatch it should be `{:dispatch []}`

Reply:
@flexsurfer is correct, dispatch is a side-effect, and it's not actually necessary here since you can call the function directly (you'll have to move the code though because request-uri-parsed is not defined at this point)

Reply:
Resolved in latest commit.